plugins/org.jkiss.dbeaver.data.transfer.ui/src/org/jkiss/dbeaver/tools/transfer/ui/pages/database/ConfigureMetadataStructureDialog.java

```diff
@@ -0,0 +1,420 @@
+import org.jkiss.dbeaver.tools.transfer.database.DatabaseMappingType;
+import org.jkiss.dbeaver.tools.transfer.database.DatabaseTransferUtils;
+import org.jkiss.dbeaver.tools.transfer.internal.DTMessages;
+import org.jkiss.dbeaver.tools.transfer.ui.internal.DTUIMessages;
+import org.jkiss.dbeaver.tools.transfer.ui.wizard.DataTransferWizard;
+import org.jkiss.dbeaver.ui.UIUtils;
+import org.jkiss.dbeaver.ui.dialogs.BaseDialog;
+import org.jkiss.dbeaver.ui.properties.PropertyTreeViewer;
+import org.jkiss.utils.ArrayUtils;
+import org.jkiss.utils.CommonUtils;
+
+import java.lang.reflect.InvocationTargetException;
+import java.util.ArrayList;
+import java.util.Map;
+
+/**
+ *  Dialog with tabs to change target table properties and table columns mapping
+ */
+public class ConfigureMetadataStructureDialog extends BaseDialog {
+
+    private DataTransferWizard wizard;
+    private DatabaseConsumerSettings settings;
+    private DatabaseMappingContainer mapping;
+    private DBSObject tableObject;
+    private TabFolder configTabs;
+    private final DatabaseConsumerPageMapping pageMapping;
+    private UIServiceSQL serviceSQL;
+    private Object sqlPanel;
+    private DBEPersistAction[] persistActions;
+    private boolean ddlTabNeedRefresh = true;
+
+    private PropertySourceEditable propertySource;
+
+    public ConfigureMetadataStructureDialog(@NotNull DataTransferWizard wizard,
+                                            @NotNull DatabaseConsumerSettings settings,
+                                            @NotNull DatabaseMappingContainer mapping,
+                                            @NotNull DatabaseConsumerPageMapping pageMapping) {
+        super(wizard.getShell(), DTUIMessages.page_configure_metadata_title, null);
+        this.wizard = wizard;
+        this.settings = settings;
+        this.mapping = mapping;
+        this.pageMapping = pageMapping;
+    }
+
+    @Override
+    protected Composite createDialogArea(Composite parent) {
+        Composite composite = super.createDialogArea(parent);
+        GridData gd = new GridData(GridData.FILL_BOTH);
+        composite.setLayoutData(gd);
+
+        configTabs = new TabFolder(composite, SWT.TOP | SWT.FLAT);
+        configTabs.setLayoutData(new GridData(GridData.FILL_BOTH));
+
+        TabItem columnsMappingTab = new TabItem(configTabs, SWT.NONE);
+        columnsMappingTab.setText(DTUIMessages.columns_mapping_dialog_shell_text);
+        ColumnsMappingDialog columnsMappingDialog = new ColumnsMappingDialog(wizard, settings, mapping);
+        columnsMappingDialog.createControl(configTabs);
+        columnsMappingTab.setData(columnsMappingDialog);
+        Control pageControl = columnsMappingDialog.getControl();
+        columnsMappingTab.setControl(pageControl);
+
+        if (mapping.getMappingType() != DatabaseMappingType.create
+            && mapping.getMappingType() != DatabaseMappingType.recreate) {
```

Comment:
Perhaps `DatabaseMappingType` deserves a method for checking for these two? There are already several occurrences of the same pattern that involves checking for both `create` and `recreate` variants.